### PR TITLE
[Tooling] Introduce `publish_release` lane

### PR DESCRIPTION
**Pull request by AliSoftware**
> [!IMPORTANT]
> This PR builds on top of https://github.com/Automattic/pocket-casts-ios/pull/2553 so that we could make those changes target `release/7.79`—in case we want to use the occasion of the `7.79` release to test this automation.

## Description

Introduce a `publish_release` lane to automate:
 - The publication of the GitHub Release of the final build (once the version submitted to Google on Friday have been approved)
 - The creation of the final backmerge PR if needed(†)
 - The deletion of the release branch

<details><summary>(†) Note that in 95% of cases there won't be anything to backmerge—and the creation of that backmerge PR will be automatically skipped.</summary>

 - This is because there had already been a backmerge PR created during submission on Friday and no new commit should have been added to the `release/*` branch after submission.
  - But calling `create_release_backmerge_pull_request` anyway is more of safety net in case there would have been a new  commit on `release/*` anyway since, to make sure not lose it completely accidentally once we delete the `release/*` branch
  - Rare cases where this could happen is if a commit was added to change tooling-related files (`Fastlane`, `Gemfile.lock`, …) after the submission to fix issues encountered in the tooling, or update of the `CHANGELOG.md` file if one notice after having submitted the build that a PR was forgotten to be added to it, etc. Those seldom cases would not require a resubmission of the final build so that'd be ok for them to land even after the submission was done, and in those rare cases a backmerge PR would be created to propagate those properly instead of risking to lose those changes upon `release/*` branch deletion
</details>

This PR also renames `GHHELPER_REPO` constant to `GITHUB_REPO`, for consistency with our other repos

## Testing Instructions

Won't really be able to test those without doing a real final release publication as part of a Release Scenario, given this lane creates side-effects.

> [!NOTE]
> I've made this PR target `release/7.79` so that we could use the occasion of the `7.79` release to test that this lane works as expected if we want (i.e. running `bundle exec fastlane publish_release` instead of doing the "Publish GitHub Release" and "Delete release branch" tasks manually) — cc @SergioEstevao 

## Next Steps

 - I will create the corresponding diff for the Release Scenario to replace the corresponding manual tasks during "Release" milestone to just call this new lane instead.
 - I will probably consider improving that lane in a future PR to also include the post of a message to Slack(?)
 - We could also consider in the future making that lane also automate the actual release of the app in AppStoreConnect. That being said, so far for others products we've kept that particular step manual given its important impact and the fact that we don't want to accidentally release a final version unexpectedly but prefer having a human having to think twice about it.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/67">7.79 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SergioEstevao — APPROVED** [review]
:shipit:

**Comment by AliSoftware** [comment]
@SergioEstevao given your :shipit: on this PR, I've now merged https://github.com/Automattic/pocket-casts-ios/pull/2553 + this PR into `release/7.79` 🙂  Which means that during the rest of your Release Manager duties for `7.79`:

 - The distribution to external testers should now be automated for the next betas (and final release) you'll do. So you should not need to do the "Add External Testers in ASC" manual task from the scenario anymore.
 - Once Apple will have had approved the final `7.79` build to be published to the App Store, you can try running `bundle exec fastlane publish_release` instead of doing the "Publish the GitHub Release" and "delete the release branch" tasks that are listed in the current `7.79` Release Scenario in MC, to test this new lane out.

_If all works as expected, I'll then plan to update the MC scenario template when I'll be back in January, to replace the manual tasks about publish GH Release + delete branch" with `publish_release` lane (hopefully before your team creates the new scenario for `7.80`)_

**Pull request merged**
